hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ReceiverTasklet.java

```diff
@@ -208,9 +217,14 @@ public int updateAndGetSendSeqLimitCompressed() {
      *
      * @param timestampNow value of the timestamp at the time the method is called. The timestamp
      *                     must be obtained from {@code System.nanoTime()}.
+     * @param expectedConnection The connection to which the result will be sent. We use it
+     *                           to check that it's the same connection the tasklet was crated with.
      */
     // Invoked sequentially by a task scheduler
-    int updateAndGetSendSeqLimitCompressed(long timestampNow) {
+    int updateAndGetSendSeqLimitCompressed(long timestampNow, Connection expectedConnection) {
+        if (!Objects.equals(expectedConnection, memberConnection)) {
+            connectionChanged = true;
```

Comment:
Does this just make a problem less likely? Since the read and write of this variable are concurrent to each other?

Reply:
It should prevent the problem completely. The `connectionChanged` is checked in a loop so if it's changed to `true` (and it's only ever changed to `true`), it will be noticed, the error will be thrown and the job will fail and not get stuck.